.github/workflows/yocto-build-deploy.yml

```diff
@@ -293,19 +294,30 @@ jobs:
       #     regex: '^v20[0-9][0-9].[0-1]?[1470].[0-9]+$'
 
       # This will control the deployment of the hostapp only - it will determine if it is marked as final or not
-      # The hostapp being finalised is what determines if the API will present this OS version to users
+      # The hostapp being finalised is what determines if the API will present this OS version to user
+      # If the test_matrix is empty - it means there are no tests for the DT - so don't check tests, and don't finalise
+      # This check will check for a workflow with the same name as the yocto machine - so ensure that workflows are named after the yocto-machine
       - name: Check test results
-        if: ${{ inputs.check-merge-tests }} # Left in the case of manual deploys where tests are failing but we had to force merge
+        if: ${{ inputs.check-merge-tests && inputs.test_matrix != '{}' }} # Left in the case of manual deploys where tests are failing but we had to force merge
```

Comment:
The input test matrix won't be empty; we have a default set for it.
So, on merge, it will have the default value. 

Or before this PR, we merge the poll option that we all agreed on `Default to no tests unless provided via `test_matrix` JSON and deprecate `run_tests` boolean`